scripts/validate_min_versions_in_sync.py

```diff
@@ -41,6 +46,85 @@
 import _optional
 
 
+def pin_min_versions_to_ci_deps():
+    all_yaml_files = list(YAML_PATH.iterdir())
+    all_yaml_files.append(ENV_PATH)
+    for curr_file in all_yaml_files:
+        with open(curr_file) as yaml_f:
+            data = update_yaml_file_version(yaml_f)
+            os.remove(curr_file)
+            with open(curr_file, "w") as f:
+                f.write(data)
+
+
+def update_yaml_file_version(yaml_file):
+    exclusion_list = {
+        "python=3.8[build=*_pypy]": None,
+    }
+    toml_dependencies = get_versions_from_toml()
+    data = yaml_file.read()
+    yaml_f = yaml.safe_load(data)
+    yaml_deps = yaml_f["dependencies"]
+    res = []
+    [res.append(x) for x in yaml_deps if x not in res]
+    yaml_deps = res
+    for dep_line in yaml_deps:
+        replace_text = operator = yaml_left = ""
+        search_text = str(dep_line)
+        if str(dep_line) in exclusion_list:
+            continue
+        if ">=" in dep_line:
+            operator = ">="
+        elif "==" in dep_line:
+            operator = "=="
+        elif "=" in dep_line:
+            operator = "="
+        elif "<" in dep_line:
+            operator = "<"
+        elif ">" in dep_line:
+            operator = ">"
+        else:
+            operator = ""
+        if operator == "":
+            yaml_package, yaml_version = str(dep_line).strip(), None
+            yaml_left = yaml_package
+        else:
+            yaml_package, yaml_version = str(dep_line).strip().split(operator)
+            if operator == "<" or operator == ">":
```

Comment:
pylint is suggesting to rewrite this one
```
 pylint...................................................................Failed
- hook id: pylint
- duration: 541.86s
- exit code: 8

************* Module scripts.validate_min_versions_in_sync
scripts/validate_min_versions_in_sync.py:93:15: R1714: Consider merging these comparisons with 'in' by using 'operator in ('<', '>')'. Use a set instead if elements are hashable. (consider-using-in)
```

(pylint is very slow, so it currently only runs in CI - but `ruff` is re-implementing a large portion of its checks, hopefully we can remove it and just use ruff soon-ish, which is blazingly fast)